interfaces/portal/src/app/pages/project-payment-transfer-list/components/retry-transfers-dialog/retry-transfers-dialog.component.ts

```diff
@@ -66,14 +72,33 @@ export class RetryTransfersDialogComponent {
   }));
 
   public retryFailedTransfers({ referenceIds }: { referenceIds: string[] }) {
+    const eventName =
+      referenceIds.length === 1
+        ? 'retry-transaction:single'
+        : 'retry-transaction:multiple';
+    this.trackingService.trackEvent({
+      category: TrackingCategory.manageTransactions,
+      action: TrackingAction.clickRetryTransactionButton,
+      name: eventName,
+      value: referenceIds.length,
+    });
+
     if (this.paymentStatus.data()?.inProgress) {
       this.toastService.showToast({
         severity: 'warn',
         detail: $localize`A payment is currently in progress. Please wait until it has finished.`,
       });
       return;
     }
+
     this.referenceIdsForRetryTransfers.set(referenceIds);
-    this.retryTransfersConfirmationDialog().show();
+    this.retryTransfersConfirmationDialog().show({
+      trackingEvent: {
+        category: TrackingCategory.manageTransactions,
+        action: TrackingAction.clickProceedButton,
+        name: eventName,
+        value: referenceIds.length,
+      },
+    });
```

Comment:
how does it work to put this as a parameter to .show(). Does that mean this event is tracked upon showing? That is something else as clicking the proceed button, which is what the action implies?

Reply:
OK, I see it in the (pre-existing) code now indeed. Not very clear without knowing that, but fine for now.